### PR TITLE
Add camel-atmosphere-websocket wrapper

### PR DESCRIPTION
**Pull request by lopushen**

### CODE REVIEW

File: features/src/main/feature/camel-features.xml [hunk 1]
```diff
@@ -302,6 +302,12 @@
         <bundle dependency='true'>mvn:org.asteriskjava/asterisk-java/${asterisk-java-version}</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-asterisk/${project.version}</bundle>
     </feature>
+    <feature name='camel-atmosphere-websocket' version='${project.version}' start-level='50'>
+        <feature version='${camel.osgi.version.range}'>camel-servlet</feature>
+        <feature prerequisite="true">spifly</feature>
```
Comment (lopushen) [inline features/src/main/feature/camel-features.xml:307]: This is to resolve
`[camel-atmosphere-websocket/4.5.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.atmosphere.websocket)(version>=3.0.0)(!(version>=4.0.0)))" [caused by: Unable to resolve org.atmosphere.runtime/3.0.4: missing requirement [org.atmosphere.runtime/3.0.4] osgi.extender; filter:="(osgi.extender=osgi.serviceloader.processor)"]]]
`
same as in `camel-openstack`

File: features/src/main/feature/camel-features.xml [hunk 2]
```diff
@@ -302,6 +302,12 @@
...
+        <bundle dependency="true">wrap:mvn:org.atmosphere/atmosphere-runtime/${atmosphere-version}$overwrite=merge&amp;Import-Package=jakarta.servlet;version:="[6,7)",*;resolution:=optional</bundle>
```
Comment (lopushen) [inline features/src/main/feature/camel-features.xml:309]: The original (with no wrap) import is 
```
Import-Package = 
	org.atmosphere.inject;resolution:=optional,
	org.atmosphere.inject.annotation;resolution:=optional,
	jakarta.servlet;version="[2.5,4)",
	jakarta.servlet.annotation;version="[2.5,4)",
	jakarta.servlet.http;version="[2.5,4)",
	jakarta.websocket;resolution:=optional;version="[2.0,3)",
	jakarta.websocket.server;resolution:=optional;version="[2.0,3)",
	jakarta.enterprise.context;resolution:=optional;version="[3.0,4)",
	jakarta.enterprise.inject;resolution:=optional;version="[3.0,4)",
	jakarta.inject;resolution:=optional;version="[2.0,3)",
	io.quarkus.runtime;resolution:=optional,
	org.atmosphere.jersey.util;resolution:=optional,
	org.testng;resolution:=optional,
	jakarta.annotation;version="[2.0,3)",
	javax.xml.parsers,
	org.apache.commons.pool2;resolution:=optional;version="[2.3,3)",
	org.apache.commons.pool2.impl;resolution:=optional;version="[2.3,3)",
	org.atmosphere.cache,
	org.atmosphere.client,
	org.atmosphere.config,
	org.atmosphere.config.managed,
	org.atmosphere.config.service,
	org.atmosphere.container,
	org.atmosphere.container.version,
	org.atmosphere.handler,
	org.atmosphere.interceptor,
	org.atmosphere.lifecycle,
	org.atmosphere.pool,
	org.atmosphere.util,
	org.atmosphere.util.analytics,
	org.atmosphere.util.annotation,
	org.atmosphere.util.uri,
	org.atmosphere.websocket.protocol,
	org.json;resolution:=optional;version="[20211205.0,20211206)",
	org.slf4j;version="[1.7,2)",
	org.w3c.dom,
	org.xml.sax,
	org.eclipse.jetty.continuation;resolution:=optional;version="[7.6,10)",
	org.eclipse.jetty.websocket;resolution:=optional;version="[7.6,10)"

```
So apart from jakarta all the non-optional imports are either from the actual atmosphere-runtime jar or present in the JDK. So, forcing the already used jakarta and making all the other imports optional

**Review by f2par0 — APPROVED** [review]
Ok for me, @essobedo  we chose to force upgrade the version of jarkarta , but if you have a better solution ....

- no conflict :heavy_check_mark: 
- compilation works :heavy_check_mark: 
- no unnecessary usage of wrap protocol :heavy_check_mark: 
- no unnecessary dependencies :heavy_check_mark: 
- no hardcoded version if it's in the parent pom :heavy_check_mark: 
- feature is placed in alphabetical order in the file camel-features.xml  :heavy_check_mark: 
- usage of dependent features (netty, jackson, jakarta ...) :heavy_check_mark:

File: features/src/main/feature/camel-features.xml (hunk 2 above)
Comment (essobedo) [reply features/src/main/feature/camel-features.xml:309]: AFAIU from the [pom of the project itself](https://github.com/Atmosphere/atmosphere/blob/main/modules/cpr/pom.xml#L101-L104), the version range should rather be `[5,6)`

Comment (lopushen) [reply features/src/main/feature/camel-features.xml:309]: You're right, and we will end up with 2 jakarta features
The component requites camel-servlet, which needs jakarta 6
```
[INFO] --- maven-dependency-plugin:3.6.1:tree (default-cli) @ camel-atmosphere-websocket ---
[INFO] org.apache.camel:camel-atmosphere-websocket:jar:4.5.1-SNAPSHOT
[INFO] +- org.apache.camel:camel-support:jar:4.5.1-SNAPSHOT:compile
[INFO] |  +- org.apache.camel:camel-api:jar:4.5.1-SNAPSHOT:compile
[INFO] |  |  \- jakarta.xml.bind:jakarta.xml.bind-api:jar:4.0.2:compile
[INFO] |  |     \- jakarta.activation:jakarta.activation-api:jar:2.1.3:compile
[INFO] |  +- org.apache.camel:camel-management-api:jar:4.5.1-SNAPSHOT:compile
[INFO] |  +- org.apache.camel:camel-util:jar:4.5.1-SNAPSHOT:compile
[INFO] |  +- org.apache.camel:camel-util-json:jar:4.5.1-SNAPSHOT:compile
[INFO] |  +- org.apache.camel:camel-xml-jaxp-util:jar:4.5.1-SNAPSHOT:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:2.0.12:compile
[INFO] +- org.apache.camel:camel-servlet:jar:4.5.1-SNAPSHOT:compile
[INFO] +- jakarta.servlet:jakarta.servlet-api:jar:6.0.0:compile
[INFO] \- org.atmosphere:atmosphere-runtime:jar:3.0.4:compile
```

Comment (essobedo) [reply features/src/main/feature/camel-features.xml:309]: Please note that we already have 2 features for the servlet API but ok let's keep it like this for now
```
    <feature name="jakarta-servlet" version="5.0.0">
        <bundle dependency="true">mvn:jakarta.servlet/jakarta.servlet-api/5.0.0</bundle>
    </feature>

    <feature name="jakarta-servlet" version="6.0.0">
        <bundle dependency="true">mvn:jakarta.servlet/jakarta.servlet-api/6.0.0</bundle>
    </feature>
```

Comment (essobedo) [inline features/src/main/feature/camel-features.xml:309]:
```suggestion
        <!-- Wrap protocol used to workaround the wrong version range used for the servlet API [2.5,4)  -->
        <bundle dependency="true">wrap:mvn:org.atmosphere/atmosphere-runtime/${atmosphere-version}$overwrite=merge&amp;Import-Package=jakarta.servlet;version:="[6,7)",*;resolution:=optional</bundle>
```

File: features/src/main/feature/camel-features.xml [hunk 3]
```diff
@@ -302,6 +302,13 @@
         <bundle dependency='true'>mvn:org.asteriskjava/asterisk-java/${asterisk-java-version}</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-asterisk/${project.version}</bundle>
     </feature>
+    <feature name='camel-atmosphere-websocket' version='${project.version}' start-level='50'>
+        <feature version='${camel.osgi.version.range}'>camel-servlet</feature>
+        <feature prerequisite="true">spifly</feature>
+        <!-- Wrap protocol used to workaround the wrong version range used for the servlet API [2.5,4)  -->
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <!-- Wrap protocol used to work around the wrong version range used for the servlet API [2.5,4)  -->
```

File: features/src/main/feature/camel-features.xml (hunk 1 above)
Comment (essobedo) [reply features/src/main/feature/camel-features.xml:307]: Surprisingly it compiles without that

Comment (essobedo) [reply features/src/main/feature/camel-features.xml:307]: My bad, it is indeed required

**Review by essobedo — APPROVED** [review]
LGTM

**Pull request merged**